pkg/util/ed25519.go

```diff
@@ -0,0 +1,31 @@
+package util
+
+import (
+	"crypto/ed25519"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+)
+
+func ParseEd25519PrivateKey(pemString string) (ed25519.PrivateKey, error) {
+	if len(pemString) == 0 {
+		return nil, nil
```

Comment:
from an utility function's perspective, empty pem string is unexpected, you need to return an error